Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/expiration/DccExpirationChecker.kt

```diff
@@ -0,0 +1,38 @@
+package de.rki.coronawarnapp.covidcertificate.expiration
+
+import dagger.Reusable
+import de.rki.coronawarnapp.covidcertificate.common.certificate.CwaCovidCertificate
+import de.rki.coronawarnapp.covidcertificate.common.certificate.DccData
+import de.rki.coronawarnapp.util.TimeAndDateExtensions.daysUntil
+import org.joda.time.DateTimeZone
+import org.joda.time.Duration
+import org.joda.time.Instant
+import javax.inject.Inject
+
+@Reusable
+class DccExpirationChecker @Inject constructor() {
+
+    fun getExpirationState(
+        dccData: DccData<*>,
+        expirationThreshold: Duration,
+        now: Instant,
+        timeZone: DateTimeZone = DateTimeZone.getDefault()
+    ): CwaCovidCertificate.State = with(dccData) {
+        val expiresAt = header.expiresAt
+        val daysUntilExpiration = now.daysUntil(expiresAt, timeZone)
+
+        return when {
+            daysUntilExpiration < 0 -> CwaCovidCertificate.State.Expired(expiresAt)
+            daysUntilExpiration == 0 -> {
+                if (now.toDateTime(timeZone).isAfter(expiresAt)) {
```

Comment:
Here you can just compare millis, both are Instants. time zone should not affect this.

Reply:
Thanks, will adjust.